core/src/main/java/com/novoda/noplayer/internal/exoplayer/NoPlayerAdsLoader.java

```diff
@@ -67,9 +70,10 @@ public void start(final EventListener eventListener, AdViewProvider adViewProvid
 
     private final AdvertsLoader.Callback advertsLoadedCallback = new AdvertsLoader.Callback() {
         @Override
-        public void onAdvertsLoaded(List<AdvertBreak> advertBreaks) {
+        public void onAdvertsLoaded(List<AdvertBreak> breaks) {
             loadingAds = null;
-            adPlaybackState = AdvertPlaybackState.from(advertBreaks);
+            advertBreaks = breaks;
```

Comment:
can we instead `advertBreaks.clear(); advertBreaks.addAll(breaks)` so that the local reference stays in place

Reply:
And the field could then be final. 👍

Reply:
https://github.com/novoda/no-player/pull/208/commits/7ddefbc2641359663aa5fcde7836be5e3c08ad31